homeassistant/components/samsungtv/bridge.py

```diff
@@ -223,7 +223,8 @@ def try_connect(self):
                 return RESULT_SUCCESS
             except WebSocketException:
                 LOGGER.debug("Working but unsupported config: %s", config)
-                return RESULT_NOT_SUPPORTED
+                if self.port != 8001:
```

Comment:
Does it matter if it's 8001 specifically or just that it's the first round in the iteration? I'd change to check the iteration round if that's what's important. Otherwise we're making this dependent on the order of ports in the iteration.

Reply:
Yes, that's exactly what I was trying, but no nice python'ic solution.

Reply:
I think we can use `for` ... `else`. https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops

```py
result = None
for self.port in (8001, 8002):
    ...
    except WebSocketException:
        result = RESULT_NOT_SUPPORTED
    ...
else:
    if result:
        return result

return RESULT_NOT_SUCCESSFUL
```

Reply:
Nice.